src/components/with-newsletter-api/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { dispatch, select } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useState } from '@wordpress/element';
+
+export default () =>
+	createHigherOrderComponent(
+		OriginalComponent => props => {
+			const [ inFlight, setInFlight ] = useState( false );
+			const { createSuccessNotice, createErrorNotice, removeNotice } = dispatch( 'core/notices' );
+			const { getNotices } = select( 'core/notices' );
+			const apiFetchWithErrorHandling = apiRequest => {
+				setInFlight( true );
+				return new Promise( ( resolve, reject ) => {
+					apiFetch( apiRequest )
+						.then( response => {
+							const { message } = response;
+							if ( message ) {
+								getNotices().forEach( notice => removeNotice( notice.id ) );
+								createSuccessNotice( message );
+							}
+							setInFlight( false );
+							resolve( response );
+						} )
+						.catch( error => {
+							const { message } = error;
+							getNotices().forEach( notice => removeNotice( notice.id ) );
+							createErrorNotice( message );
+							setInFlight( false );
+							reject( error );
+						} );
+				} );
+			};
+			return (
+				<OriginalComponent
+					{ ...props }
+					apiFetchWithErrorHandling={ apiFetchWithErrorHandling }
+					inFlight={ inFlight }
+				/>
+			);
+		},
+		'newspack-newsletters-with-api-response'
```

Comment:
This should match the file name. And I'd change it to `with-api-handler`, as there's nothing newsletterish about this HOC – however there could be, if the HOC itself would set the API prefix (`/newspack-newsletters/v1`)

Reply:
Fixed in https://github.com/Automattic/newspack-newsletters/pull/104/commits/125b52ae8cbe6c361ed2eb6e34de13c397214fc4, unless you meant to remove the prefix as well (`newspack-newsletters-`)?

Reply:
https://github.com/Automattic/newspack-newsletters/pull/104/commits/a914a4255bea6ede1f9db55eb3cbbc27e5e9847d